pcsx2/Sio.cpp

```diff
@@ -120,13 +120,26 @@ void sioInit()
 
 bool isR3000ATest = false;
 
+u16 currentStat = 0, lastStat = 0, currentMode = 0, lastMode = 0, currentBaud = 0, lastBaud = 0;
+
 // Check the active game's type, and fire the matching interrupt.
 // The 3rd bit of the HW_IFCG register lets us know if PSX mode is active. 1 = PSX, 0 = PS2
 // Note that the R3000A's call to interrupts only calls the PS2 based (lack of) delays.
 SIO_FORCEINLINE void sioInterrupt() {
 	if ((psxHu32(HW_ICFG) & (1 << 3)) && !isR3000ATest) {
-		if (!(psxRegs.interrupt & (1 << IopEvt_SIO)))
-			PSX_INT(IopEvt_SIO, 64); // PSXCLK/250000);
+		if (!(psxRegs.interrupt & (1 << IopEvt_SIO))) {
+			currentStat = sio.StatReg & 67106816; // Mask out bits 11-25
```

Comment:
**Note:** Personally I prefer hexadecimal for bitwise operations like this, seems more readable.  This could be ``0x3FFF800``which stands out as ``0011 1111 1111 1111 1000 0000 0000`` showing which bits you fetch easily without the need of comments. Whereas, ``67106816`` makes us pop up a calculator or just hope the author's comment is correct.

Reply:
Yeah, definitely use something readable here please.